src/core/utils/helpers.ts

```diff
@@ -135,3 +135,8 @@ export const dateMatchesUsFormat = (date: string | null) => {
   const returnValue = dateFound && dateFound.length > 0 ? dateFound[0] : null;
   return returnValue;
 };
+
+export function generateMapId(index: number | string) {
+  const randomId = Math.random().toString(36);
+  return randomId.concat(index.toString());
+}
```

Comment:
Please reconsider this approach. [Random does not seem to be a good approach to React keys](https://www.developerway.com/posts/react-key-attribute).

Reply:
Uuuu very interesting! I was living in the dark for so long 😝 Thanks!